app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -95,5 +99,19 @@ def shibboleth
       end
     end
   end
+
+  private
+
+  # choose_shib_attr
+  # Shibboleth may return semi-column seperated attributes.
+  # The function returns the first one if attributes_str is not nil, otherwise
+  # it returns "(blank)" as a placeholder
+  def choose_shib_attr(attributes_str)
```

Comment:
Since shibboleth attributes are unordered, we can only choose one (here it's the first) hoping it makes sense.
Choice of attribuets like school, department, or year can be improved by having a list of them in advance and compare, but I'll do it in another PR.